### PR TITLE
Jetpack Firewall Settings: Redirect users with Protect installed to the standalone plugin's settings page

### PR DESCRIPTION
**Pull request by nateweller**
Resolves https://github.com/Automattic/jetpack-scan-team/issues/1308

## Proposed changes:

* When the Protect plugin is detected on a site, show a prompt to use the Protect firewall settings dashboard instead of the Jetpack settings card.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

peb6dq-2r8-p2#comment-1717

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Spin up a JN site running this branch with Jetpack.
* Validate the firewall settings continue to display and function as expected.
* Install Jetpack Protect.
* Validate the redirect prompt is shown instead of the firewall settings in Jetpack.

## Screenshots

Screenshot 2024-07-31 at 1 58 15 PM

### CODE REVIEW

**Review by dkmyta — CHANGES_REQUESTED** [review]
Code looks good, but seems we default to displaying the new card/redirect message even when Protect is installed but inactive, then of course the link leads nowhere useful. This got me thinking, what if Protect is installed/active but somehow not in a functioning state, do we need to or can we even account for that possibility, or do we just assume that risk.

File: projects/plugins/jetpack/_inc/client/security/waf.jsx [hunk 1]
```diff
@@ -498,6 +537,10 @@ export default connect(
 		return {
 			automaticRulesAvailable: getAutomaticRulesAvailable( state ),
 			hasScan: siteHasFeature( state, 'scan' ),
+			protectAdminUrl: `${ getSiteAdminUrl( state ) }admin.php?page=jetpack-protect#/firewall`,
+			isProtectInstalled: PROTECT_PLUGIN_FILES.some( pluginFile =>
+				isPluginInstalled( state, pluginFile )
```
Comment (dkmyta) [inline projects/plugins/jetpack/_inc/client/security/waf.jsx]: Looks like there is an action to check for an active state that might be more applicable here:
```suggestion
				isPluginActive( state, pluginFile )
```

Comment (nateweller) [reply projects/plugins/jetpack/_inc/client/security/waf.jsx]: Good catch! Applied via https://github.com/Automattic/jetpack/pull/38655/commits/c63adbc9cc0716cb78722c70b5c4025d219fe7da :+1:

**Comment by nateweller** [comment]
@dkmyta I've updated the logic to ensure Protect is active, not just installed. Should be good now!

> This got me thinking, what if Protect is installed/active but somehow not in a functioning state, do we need to or can we even account for that possibility, or do we just assume that risk.

If the plugin is considered active, I think that is good enough at this time to be confident linking to it. :+1:

**Review by dkmyta — APPROVED** [review]
Works great!

**Pull request merged**